ts/features/idpay/initiative/details/components/InitiativeBonusCounter.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from "react";
+import { StyleSheet, View } from "react-native";
+import Animated, {
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming
+} from "react-native-reanimated";
+import { VSpacer } from "../../../../../components/core/spacer/Spacer";
+import { H1 } from "../../../../../components/core/typography/H1";
+import { LabelSmall } from "../../../../../components/core/typography/LabelSmall";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import { formatNumberAmount } from "../../../../../utils/stringBuilder";
+import { Skeleton } from "../../../common/components/Skeleton";
+
+type CounterType = "Amount" | "AmountWithProgress";
+
+type BaseProps = {
+  type: CounterType;
+  isDisabled?: boolean;
+  label?: string;
+};
+
+type AmountProps = {
+  type: "Amount";
+  amount: number;
+};
+
+type AmountWithProgressProps = {
+  type: "AmountWithProgress";
+  amount: number;
+  progress: number;
+};
+
+export type BonusCounter =
+  | BaseProps &
```

Comment:
```suggestion
   BaseProps &
```
why the union type symbol here?

Reply:
New lines in type definition requires `|`.